pom.xml

```diff
@@ -2132,5 +2126,23 @@
                 </pluginManagement>
             </build>
         </profile>
+        <profile>
+            <!-- Signs build artifacts if a GPG key is configured. This is
+            required when deploying a release, and optional otherwise. -->
+            <id>sign</id>
+            <activation>
+                <property>
+                    <name>env.MAVEN_GPG_KEY</name>
+                </property>
+            </activation>
```

Comment:
This way the profile is activated during the GitHub Actions deploy step and when I run the build locally, but not otherwise.